File: codility/lessons/2_arrays/odd_occurrences_in_array.c
// https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/

int solution(int A[], int N) {
    int number = 0;
    for (int i = 0; i < N; i++) {
        number ^= A[i];
    }
    return number;
}

File: codility/lessons/2_arrays/odd_occurrences_in_array.py
# https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/

from collections import Counter

def solution(A):
    occurrences = Counter(A)
    for (value, count) in occurrences.items():
        if count%2 == 1:
            return value
    return -1
